components/brave_wallet_ui/common/async/base-query-cache.ts

```diff
@@ -483,6 +487,43 @@ export class BaseQueryCache {
     return this._nftMetadataRegistry[tokenId]
   }
 
+  getSpamNftsForAccountId = async (accountId: BraveWallet.AccountId) => {
+    if (!this.spamNftsForAccountRegistry[accountId.uniqueKey]) {
+      const { braveWalletService } = getAPIProxy()
+      const { address, coin } = accountId
+      const networksRegistry = await cache.getNetworksRegistry()
+
+      const chainIds = networksRegistry.ids.map(
+        (network) => networksRegistry.entities[network]!.chainId
+      )
+
+      let currentCursor: string | null = null
+      const accountSpamNfts = []
+
+      do {
+        const {
+          tokens,
+          cursor
+        }: {
+          tokens: BraveWallet.BlockchainToken[]
+          cursor: string | null
+        } = await braveWalletService.getSimpleHashSpamNFTs(
+          address,
+          chainIds,
+          coin,
+          currentCursor
+        )
+
+        accountSpamNfts.push(...tokens)
+        currentCursor = cursor
+      } while (currentCursor)
+
+      this.spamNftsForAccountRegistry[accountId.uniqueKey] = accountSpamNfts
+    }
```

Comment:
This just fetches everything with cursor. Should we do that in backend instead?

Reply:
We could if you would like to take a look are porting this to core

Reply:
@supermassive  Do we want to handle this in this PR or in a follow-up?

Reply:
Some another PR would be ok

Reply:
Logged: https://github.com/brave/brave-browser/issues/38823